### ISSUE TITLE
ocf_run mangles output, may expand shell globs or worse, causing misleading (at best) logs

### ISSUE THREAD

**Issue by lge** [ClusterLabs/resource-agents#930]
See: https://github.com/ClusterLabs/resource-agents/blob/master/heartbeat/ocf-shellfuncs.in#L436

```
output=`"$@" 2>&1`
rc=$?
output=`echo $output`
```

can someone remember what the second assignment is for?

If the output contains (e.g.) "*** some three stars ***",
the `echo $output` subshell will treat this as wildcards,
and unless executed in an empty directory, will cause confusing messages to be logged.
(seen in the wild: a number of expanded core.12345 files that happened to still be around)

If the only purpose was to "squeeze blanks and newlines",
how about
```
output=`echo "$output" | tr -s ' \t\r\n' ' '`
```

**Comment by dmuhamedagic** [comment]
On Wed, Feb 22, 2017 at 01:37:09AM -0800, Lars Ellenberg wrote:
> See: https://github.com/ClusterLabs/resource-agents/blob/master/heartbeat/ocf-shellfuncs.in#L436
> 
> ```
> output=`"$@" 2>&1`
> rc=$?
> output=`echo $output`
> ```
> 
> can someone remember what the second assignment is for?

I'd also assume that it was to get rid of the newlines.

> If the output contains (e.g.) "*** some three stars ***",
> the `echo $output` subshell will treat this as wildcards,
> and unless executed in an empty directory, will cause confusing messages to be logged.
> (seen in the wild: a number of expanded core.12345 files that happened to still be around)

Right.

> If the only purpose was to "squeeze blanks and newlines",
> how about
> ```
> output=`echo "$output" | tr -s ' \t\r\n' ' '`
> ```

Much better indeed.

**Comment by lge** [comment]
closed by dcce6a1 ocf-shellfuncs: ocf_run: avoid shell glob expansion of log messages
